crates/swc_ecma_parser/src/parser/pat.rs

```diff
@@ -643,15 +641,22 @@ impl<'a, I: Tokens> Parser<I> {
                                 },
 
                                 PropOrSpread::Spread(SpreadElement { dot3_token, expr }) => {
-                                    Ok(ObjectPatProp::Rest(RestPat {
-                                        span,
-                                        dot3_token,
-                                        // FIXME: is BindingPat correct?
-                                        arg: Box::new(
-                                            self.reparse_expr_as_pat(PatType::BindingPat, expr)?,
-                                        ),
-                                        type_ann: None,
-                                    }))
+                                    if idx != len - 1 && self.syntax().early_errors() {
+                                        syntax_error!(self, span, SyntaxError::NonLastRestParam)
```

Comment:
I think recovering from this error would be trivial.
Isn't it better to recover here?

Reply:
What recover?

Reply:
I meant using `emit_err` instead of `syntax_error!`.
With it, we can report more errors in a single invokation.

Reply:
Is new commit ok?